engine/src/main/java/io/camunda/zeebe/process/test/engine/CommandWriter.java

```diff
@@ -22,14 +23,14 @@ record CommandWriter(LogStreamWriter writer) {
   void writeCommandWithKey(
       final Long key, final UnifiedRecordValue command, final RecordMetadata recordMetadata) {
     synchronized (writer) {
-      writer.tryWrite(LogAppendEntry.of(key, recordMetadata, command));
+      writer.tryWrite(WriteContext.userCommand(), LogAppendEntry.of(key, recordMetadata, command));
```

Comment:
❓ Does it make sense to you @oleschoenburg ? I tried to understand the `WriteContext` usage

Reply:
I think so :+1: If all writes are effectively user commands then it's perfect. Even if not, I don't think using the wrong `WriteContext` would matter for ZPT as it will only be used for flow control which we ignore here AFAIK.